service/policy/db/query.sql

```diff
@@ -29,6 +29,55 @@ RETURNING id;
 -- name: DeleteKeyAccessServer :execrows
 DELETE FROM key_access_servers WHERE id = $1;
 
+---------------------------------------------------------------- 
+-- ATTRIBUTE FQN
+----------------------------------------------------------------
+
+-- name: UpsertAttributeValueFqn :one
+INSERT INTO attribute_fqns (namespace_id, attribute_id, value_id, fqn)
+SELECT
+    n.id,
+    ad.id,
+    av.id,
+    CONCAT('https://', n.name, '/attr/', ad.name, '/value/', av.value) AS fqn
+FROM attribute_namespaces n
+JOIN attribute_definitions ad ON n.id = ad.namespace_id
+JOIN attribute_values av ON ad.id = av.attribute_definition_id
+WHERE av.id = $1
+ON CONFLICT (namespace_id, attribute_id, value_id) 
+    DO UPDATE 
+        SET fqn = EXCLUDED.fqn
+RETURNING fqn;
+
+-- name: UpsertAttributeDefinitionFqn :one
+INSERT INTO attribute_fqns (namespace_id, attribute_id, value_id, fqn)
+SELECT
+    n.id,
+    ad.id,
+    NULL,
```

Comment:
They are artifacts of the squirrel queries, from the switch statement cases in the `upsertAttrFqnSQL` function.  I literally just copied them as-is over to SQL as separate queries for definition, value, and namespace.

Definition squirrel query: https://github.com/opentdf/platform/pull/1445/files#diff-6dfc9f9359c6ea6dbaf8df5306e47732e636997b51bd20ef1eaef77f6d39823dL47-L48

Namespace squirrel query: https://github.com/opentdf/platform/pull/1445/files#diff-6dfc9f9359c6ea6dbaf8df5306e47732e636997b51bd20ef1eaef77f6d39823dL52-L53

If you'd like, we could discuss further any additional cleanup/improvements you'd like to make while I'm in this code.

Reply:
Yeah this looks like a defect. I just copied your SQLC queries locally and found that:
If the definition name changes, the `UpsertAttributeDefinitionFqn` query only updates the FQN that ends in the definition, not any FQNs where the value is appended at the end.

![Screenshot 2024-10-02 at 8 22 12 AM](https://github.com/user-attachments/assets/a32dba46-c8a4-4ab8-bbfd-2c764716ee3b)

This appears to have been an existing defect or an artifact of how we treat re-indexing, but I think we should fix this now so that a reindexing upsertion for a definition updates FQNs that also include the value FQNs containing the definition change, and upsertion for a namespace updates FQNs for definitions and values that also include the namespace.

Reply:
Thinking through this more, if re-indexing as a "job" always runs all three queries with a roundtrip for each attribute object ID, this case would only matter should we get a failure partway through, which is why it hasn't been an issue. I do think we could clean this up further so the queries aren't driven by a single attribute object ID and rather drive from the entire column at once batch-style (just removing the WHERE clause I think?), but that can be a further improvement in a separate task so we don't block this improvement you're introducing.

Reply:
https://github.com/opentdf/platform/issues/1603 Handling in separate issue.

Reply:
Ticket created for this discussion: https://github.com/opentdf/platform/issues/1603